app/controllers/checkout_controller.rb

```diff
@@ -165,15 +165,6 @@ def update_failed
     end
   end
 
-  # When we have a pickup Shipping Method,
-  #   we clone the distributor address into ship_address before_save
-  # We don't want this data in the form, so we clear it out
```

Comment:
so we'll miss this behavior, right? users will see the address of the distributor which on the other hand seems good.

Reply:
the behaviour is the same but instead of an empty address we inject the distributor address (instead of this clear_ship_address, we are calling OrderUpdater#shipping_address_from_distributor).

When the ship method is pick up, the ship address form is not shown on checkout, so the user will not see it.

One test scenario I did here (will move it to the description so testers can try):
- ship method is pickup, no ship address form 
- checkout with failed payment and see payment error
- go back to ship methods and change to some delivery ship method
- make sure the ship address shown by default is not the distributor's address

Reply:
I was wondering about this case:

- ship method requires address
- checkout with failed payment
- what is the ship address now?

Reply:
yes, I tested that. the ship address is not overritten, it will be whatever the user wrote  because the shipping_address_from_distributor will not update ship address if order.shipping_method.require_ship_address